packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,296 @@
+| `trailing` | `Badge, Link, Text, IconButton`, `undefined` | undefined  |     trailing component to be placed at the right most side of the BottomSheetHeader     |    |
+| `titleSuffix` | `Counter` | `undefined` | A component to be placed adjacent to the title text |          |
+| `hideDivider` | `boolean` | `false` | Show or hide the divider |          |
+| `showBackButton` | `boolean` | `false` | Show or hide back button |          |
+| `onBackButtonClick` | `boolean` | `false` | Event handler for the back button |          |
+
+### `BottomSheetFooter` API
+
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| `children` | `React.ReactNode` | `undefined` | Contents of the BottomFooter | ✅       |
+
+## Composition with Dropdown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+#### 1. We couple the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted. Products where users are desktop only or majority are on desktop they will get the bundle of BottomSheet (vice versa)
+- BottomSheet's state will get coupled with SelectDropdown
+- There will be a lot of interdependency of state management
+- These [usecases](https://razorpay.slack.com/archives/C01CS8YBEQZ/p1677825334856589?thread_ts=1677825092.305089&cid=C01CS8YBEQZ) will be harder to solve for without the BottomSheet being an independent component.
+
+#### 2. We expose BottomSheet as an independently component and let user do the composition
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+- BottomSheet is decoupled from the state and internal logic of SelectDropdown & can also be used independently
+- Reduced implementation complexity & [interdependency](https://www.notion.so/BottomSheet-API-Discussion-e2aa79cd45274ef280fdb998efa6b98f?pvs=4#630d5e93897b4ca0bf54f475f0cd7d6e) of state
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Dropdown as per their needs.
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+> **API Rabbit Hole**  
+> If you want to go deep into the API Rabbit Hole, We also had explored various other APIs and discussed internally about the pros and cons of each which we **[documented in this Notion doc](https://www.notion.so/BottomSheet-API-Discussion-e2aa79cd45274ef280fdb998efa6b98f?pvs=4#db6bd9826bc048bd97d24a50269fc45d)**
```

Comment:
Let's add this here in a separate md file please, this link may not work in the future.

Reply:
Done.